gh-133829: Update zipimport example to not mention Python 2.3 (#133835)
python/cpython: Doc/library/zipimport.rst

Here is an example that imports a module from a ZIP archive - note that the
:mod:`zipimport` module is not explicitly used.

.. code-block:: shell-session

>    $ unzip -l example_archive.zip

Reviewer: ```suggestion

   $ unzip -l example_archive.zip
```

   Archive:  example.zip
     Length     Date   Time    Name
    --------    ----   ----    ----
        8467  05-11-25 12:29   example.py
    --------                   -------
        8467                   1 file

.. code-block:: python

   >>> import sys
   >>> sys.path.insert(0, 'example_archive.zip')  # Add .zip file to front of path
   >>> import example
   >>> example.__file__

>    'example.zip/example.py'

Reviewer: ```suggestion
   'example_archive.zip/example.py'
```

Author: @sobolevn thats outdated. I added a commit after that to fix it. Please review

Here is an example that imports a module from a ZIP archive - note that the
:mod:`zipimport` module is not explicitly used.

.. code-block:: shell-session

>    $ unzip -l example.zip

Reviewer: Please, keep this part. It is essential to understand what is inside the `.zip` file.

Reviewer: I would also probably rename `.zip` file to `example_archive.zip` so we can be sure what `import example` actually does.

Author: Okay, gimme like 2 mins!

Author: @sobolevn Done!

>    >>> sys.path.insert(0, 'pip.pyz')  # Add .pyz file to front of path

Reviewer: this is not really clear. for example, I have no idea what `pip.pyz` is and where it comes from. while the previous case was really clear.

please, update the python version and keep the example. 

I would propose something like:

```

   $ unzip -l example.zip
   Archive:  example.zip
     Length     Date   Time    Name
    --------    ----   ----    ----
        8467  11-26-02 22:30   jwzthreading.py
    --------                   -------
        8467                   1 file
   $ python
   Python REPL
   >>> import sys
```

Reviewer 2: I suggested using `pip.pyz` because `jwzthreading` is obsolete and no longer developed, but `pip` does distribute a zipimportable file. I agree though that it's currently too complex.

Reviewer: We can replace `jwzthreading` with `example.py` and `import example` 🤔 

It won't make any difference to the example's meaning. What do you think?

Author: I side with sobolevn. using filenames such as example.zip and example.py is probably the best way to go for clarity

Author: @AA-Turner and @sobolevn  please review now with ``example.zip`` ``example.py`` and ``import example``

.. code-block:: shell-session

   $ unzip -l pip.pyz
   Archive:  pip.pyz
     Length     Date   Time    Name
    --------    ----   ----    ----
       32145  05-10-25 10:00   pip/__main__.py
    --------                   -------
       32145                   1 file
   $ ./python

>    Python 3.x

Reviewer 3: Do we even need this? It is just the repl opening line.

Reviewer 4: I don't think so.

Indeed, we might break this into two snippets: bash (show unzip) and repl Python. So we can use doctest.

Author: > I don't think so.
> 
> Indeed, we might break this into two snippets: bash (show unzip) and repl Python. So we can use doctest.

On it!

Author: @adorilson @StanFromIreland Done! Please review!

.. code-block:: shell-session

   $ unzip -l example_archive.zip
   Archive:  example_archive.zip
     Length     Date   Time    Name
    --------    ----   ----    ----
        8467  05-11-25 12:29   example.py
    --------                   -------
        8467                   1 file

.. code-block:: pycon

   >>> import sys

>    >>> sys.path.insert(0, 'example_archive.zip')  # Add .zip file to front of path

Reviewer 5: I think this works better, but I'll leave the decision up to you:
```suggestion
   >>> # Add the archive to the front of the module search path
   >>> sys.path.insert(0, 'example_archive.zip')
```

.. code-block:: shell-session

> 

Reviewer 3: ```suggestion
```
We don’t need 2 lines here.

Author: Does it matter? It would invalidate the review. Also, do you know why `.. code-block:: pycon` not `.. code-block::  python`. What is pycon in rst, and why did @sobolevn request it?

Reviewer 3: > Does it matter?

Yes.

> What is pycon in rst

pycon: python console (i.e. REPL)
python: python

   Archive:  example_archive.zip
     Length     Date   Time    Name
    --------    ----   ----    ----

>         8467  05-11-25 12:29   example.py

Reviewer 2: Shall we use a clearly example time, rather than one that will get dated?

```suggestion
        8467  01-01-00 12:34   example.py
```

Author: Probably best not so we will know (or people will notice) to update the example if we ever make breaking changes to zipimport. I don't think it is worth it, but it doesn't matter

Reviewer 2: I don't understand your comment. The code block here is for `unzip`, where we're already inventing an archive for expository purposes.

Author: My phone has _aggressive_ autocorrect and my misspelling resulted in unzip turning into zipimporter.

But on second thoughts, by keeping the time and date specific, we know what version of Python it was written in and if we _ever_ change anything even methods we keep for expository purposes someone will notice and fix it.

I also don't think it matters that much :).

Reviewer 5: > But on second thoughts, by keeping the time and date specific, we know what version of Python it was written in and if we ever change anything even methods we keep for expository purposes someone will notice and fix it.

I'm not sure I understand. If we break this (which is unlikely), we'll update the example anyway. Knowing when something was written generally doesn't have much maintenance benefit.

I think I agree with Adam -- it'd be better to make this clearer that it's an example and not supposed to be actual data.

Author: All right.

>         8467  05-11-25 12:29   example.py

Reviewer 3: I’d prefer to keep the original times, it’s like a historical artefact now…

Author: A) that doesn’t matter

b) so people know this is a new, valid example

Reviewer 3: Quoting Cicero, (those books really pay off now ;-),

> To be ignorant of what occurred before you were born is to remain always a child. For what is the worth of human life, unless it is woven into the life of our ancestors by the records of history?

As for your second point, I can reasonably doubt anyone would do that. It is incorrect anyway, you have not updated the size of supposed file, that, IMO *could* be more misleading.

Author: The only thing we care about and people will notice is the example.py in the zip. No one will care or notice that it is coincidentally the same size as jwzthreading.py. It is just there for demonstration

    --------                   -------
        8467                   1 file

> .. code-block:: python

Reviewer: ```suggestion
.. code-block:: pycon
```

Author: Wait. Pycon?

.. code-block:: shell-session

   $ unzip -l pip.pyz
   Archive:  pip.pyz
     Length     Date   Time    Name
    --------    ----   ----    ----
       32145  05-10-25 10:00   pip/__main__.py
    --------                   -------
       32145                   1 file

Now for the Python:

.. code-block:: python

> 

Reviewer 2: This is overly complex, please revert. The point is to have a simple and quick example.
